### ISSUE TITLE
touchesMoved:withEvent: shouldn't call setSelectedSegmentIndex:animated:

### ISSUE THREAD

**Issue by getaaron** [designatednerd/DNSCastroSegmentedControl#5]
The current implementation does this:

```
- (void)touchesMoved:(NSSet *)touches withEvent:(UIEvent *)event
{
    // some stuff…
    
    [self setSelectedSegmentIndex:roundedSection animated:YES];
}
```

I believe most users will consider it unexpected if an app's state changes while they're still dragging their finger. Normally, it's safe to drag controls around on the screen, and nothing will take effect until the finger is lifted. For example, in Settings, a user can drag the Airplane Mode switch right and left, and Airplane Mode won't enable unless they lift their finger while the switch is enabled.

Instead, I believe this behavior should be in `touchesEnded:`, and `touchesCancelled:` should revert `selectionView` to its original state.

**Comment by getaaron** [comment]
To be clear, it's not the UI updates I think are unexpected. I just think `UIControlEventValueChanged` shouldn't be sent until the finger is lifted.

**Comment by designatednerd** [comment]
I think you're correct. I'm swamped this week but I promise I'll get to it. :smile:

**Comment by getaaron** [comment]
No rush… and I imagine we'll both be swamped next week :)

**Comment by designatednerd** [comment]
Yuuuuuuuuuup.

**Comment by designatednerd** [comment]
Fixed in 1.3.0
